### PR TITLE
Implement PSR-7 for HTTP messages handling

### LINKED ISSUE #15507: HTTP handling should implement PSR-7
The HTTP request and HTTP response handling should be refactored to implement the [PSR-7](https://www.php-fig.org/psr/psr-7/) standard.

From the PSR-7 standard:
> This document describes common interfaces for representing HTTP messages as described in [RFC 7230](http://tools.ietf.org/html/rfc7230) and [RFC 7231](http://tools.ietf.org/html/rfc7231), and URIs for use with HTTP messages as described in [RFC 3986](http://tools.ietf.org/html/rfc3986).

### PR DESCRIPTION
**Pull request by MauricioFauth**
This PR uses [nyholm/psr7](https://github.com/Nyholm/psr7), but it could change in the future.

Fixes #15507.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
Could we have classes that extend the vendor classes to limit the exposure of a vendor in the codebase?
One of the ideas is that we could change vendors using a bunch of lines or a patch for some strange reasons or for packaging reasons :)

**Comment by mauriciofauth** [comment]
> Could we have classes that extend the vendor classes to limit the exposure of a vendor in the codebase?

PSR interfaces are great for that. For Nyholm classes, I just use it for now to make it work. The idea is to be able to switch between any PSR-7 implementation.

File: test/classes/Controllers/TransformationOverviewControllerTest.php [hunk 1]
```diff
@@ -36,17 +38,22 @@ protected function setUp(): void
 
     public function testIndexAction(): void
     {
```
Comment (williamdes) [inline test/classes/Controllers/TransformationOverviewControllerTest.php:40]: this needs to be moved to the abstract test class I created

Comment (mauriciofauth) [reply test/classes/Controllers/TransformationOverviewControllerTest.php:40]: I agree, but creating full instances is probably is not a good idea anyway.

**Pull request closed without merging**